### DISCUSSION TITLE
Trigger deferred task from a different dag context

### DISCUSSION THREAD

**Discussion by sherodtaylor** [apache/airflow#25798] [Q&A]
Currently, the way I understand is airflow is serializing the trigger tied to the dag I'm wondering if I could trigger an event for a different dag run.

I want to create a kafka consumer that has event data relating to a dag and triggering an event regardless of which deferred process its running under for example.

```
 dag 1 -> defer and create trigger -> consume from kafka  (same consumer group so split messages between each) -> receive message related to dag 2 -> yield event and trigger dag 2
 dag 2 -> defer and create trigger -> consume from kafka (same consumer group so split messages between each)

```
so they can share the same consumer group and arbitrarily trigger tasks unrelated to the dag in context? Can you change the context of the dag maybe?

Right now I'm using [airflow-provider-kafka](https://github.com/astronomer/airflow-provider-kafka/issues/9) and the problem is it's not really leveraging Kafka's ability to scale and split the consumer ideally the processor of the events could trigger different tasks associated with that?

**Comment by potiuk** [comment]
I do not really understand what you want to do. But you should likely look at https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-42+Dynamic+Task+Mapping (already implemented in 2.3) and https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-48+Data+Dependency+Management+and+Data+Driven+Scheduling (will be released in a few weeks in 2.4) and you will likely find what you need.
